src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/matcher/FilterableResolver.java

```diff
@@ -292,12 +295,27 @@ public static List<DBFilterBase> findDynamic(CaptureExpression captureExpression
 
       if (captureExpression.marker() == ALL_GROUP) {
         List<Object> arrayValue = (List<Object>) filterOperation.operand().value();
-        filters.add(new DBFilterBase.AllFilter(captureExpression.path(), arrayValue));
+        filters.add(new DBFilterBase.AllFilter(captureExpression.path(), arrayValue, false));
+      }
+
+      if (captureExpression.marker() == NOT_ANY_GROUP) {
+        List<Object> arrayValue = (List<Object>) filterOperation.operand().value();
+        filters.add(new DBFilterBase.AllFilter(captureExpression.path(), arrayValue, true));
       }
 
       if (captureExpression.marker() == SIZE_GROUP) {
         BigDecimal bigDecimal = (BigDecimal) filterOperation.operand().value();
-        filters.add(new DBFilterBase.SizeFilter(captureExpression.path(), bigDecimal.intValue()));
+        // Flipping size operator will multiply the value by -1
+        // Negative means check array_size[?] != ?
+        int size = bigDecimal.intValue();
+        DBFilterBase.MapFilterBase.Operator operator;
+        if (size > 0) {
+          operator = DBFilterBase.MapFilterBase.Operator.MAP_EQUALS;
+        } else {
```

Comment:
Ah. So here the earlier negative value for `$size` (multiply by -1) is converted to "not equals $size". Makes sense.